src/core/templates/show-people.js

```diff
@@ -62,113 +81,217 @@ const orcidIcon = () => html`<svg
+      : html`<span class="${hCard}">${company}</span>`;
+    contents.push(html` (${companyElem})`);
+  }
+  if (person.note) contents.push(document.createTextNode(` (${person.note})`));
+  if (person.extras) {
+    person.extras
+      .map(extra => html`, ${renderExtra(extra)}`)
+      .reduce((contents, html) => {
+        contents.push(html);
+        return contents;
+      }, contents);
+  }
+  if (person.retiredDate) {
+    const { retiredDate } = person;
+    const time = html`<time datetime="${retiredDate}"
+      >${humanDate(retiredDate)}</time
+    >`;
+    contents.push(html` - ${l10n.until(time)} `);
+  }
+  const dd = html`<dd
+    class="editor p-author h-card vcard"
+    data-editor-id="${editorId}"
+  >
+    ${contents}
+  </dd>`;
+  return dd;
+}
+
+function renderExtra(extra) {
+  const classVal = extra.class || null;
+  const { name, href } = extra;
+  return href
+    ? html`<a href="${href}" class="${classVal}">${name}</a>`
+    : html`<span class="${classVal}">${name}</span>`;
+}
+
+/**
+ * @param {Person} person
+ * @param {Number} index
+ * @returns
+ */
+function validatePerson(person, index) {
+  const hint =
+    "See [Person](https://respec.org/docs/#person) configuration for available options.";
+  if (!person.name) {
+    const msg = `Person object at index ${index} is missing required member "name"`;
+    showError(msg, name, { hint });
+    return false;
+  }
+
+  if (person.orcid && !validateAndCanonicalizeOrcid(person, index)) {
+    return false;
+  }
+
+  if (
+    person.retiredDate &&
+    !validateDateMember("retiredDate", person.retiredDate, index, hint)
+  ) {
+    return false;
+  }
+
+  if (person.hasOwnProperty("extras") && !validateExtras(person.extras, hint)) {
+    return false;
+  }
+
+  if (person.url && person.mailto) {
+    const msg = `Person object at index ${index} has both "url" and "mailto" property.`;
+    showWarning(msg, name, {
+      hint: `Please chose either "url" or "mailto" ("url" is preferred). ${hint}`,
+    });
+  }
+
+  if (person.companyURL && !person.company) {
+    const msg = `Person object at index ${index} has a "companyURL" member but no "company" member.`;
+    showWarning(msg, name, { hint: `Please add a "company" member. ${hint}` });
+  }
+  return true;
+}
+
+/**
+ *
+ * @param {PersonExtras} extras
+ * @param {string} hint
+ */
+function validateExtras(extras, hint) {
+  if (!Array.isArray(extras)) {
+    showError(`A person's "extras" member must be an array.`, name, { hint });
+    return false;
+  }
+  return extras.every((extra, index) => {
+    switch (true) {
+      case typeof extra !== "object":
+        showError(`"extra" index ${index} is not an object.`, name, {
+          hint,
+        });
+        return false;
+      case !extra.hasOwnProperty("name"):
+        showError(
+          `\`PersonExtra\` object at index ${index} is missing required "name" member.`,
+          name,
+          { hint }
         );
```

Comment:
This validation feels a little too much TBH.

Reply:
I don't mind it, tbh. `PersonExtra` is rarely used, as we added it to support ORCID before we added the orcid member.